src/py/mat3ra/made/tools/build/passivation/helpers.py

```diff
@@ -0,0 +1,127 @@
+        material (Material): The material to be passivated.
+        passivant (str): The chemical symbol of the passivating atom (default is 'H').
+        bond_length (float): The bond length for the passivation (default is 1.0).
+
+    Returns:
+        Material: The passivated material.
+    """
+    analyzer = SurfacePassivationMaterialAnalyzer(
+        material=material, passivant=passivant, bond_length=bond_length, shadowing_radius=shadowing_radius, depth=depth
+    )
+
+    passivant_coordinates = analyzer.get_passivant_coordinates()
+    passivant_configs = [
+        AtomAtCoordinateConfiguration(
+            crystal=material, element=AtomSchema(chemical_element=passivant), coordinate=coord
+        )
+        for coord in passivant_coordinates
+    ]
+    config = PassivationConfiguration(
+        merge_components=[material, *passivant_configs], passivant=passivant, bond_length=bond_length
+    )
+
+    builder = PassivationBuilder()
+    return builder.get_material(config)
+
+
+def passivate_dangling_bonds(
+    material: MaterialWithBuildMetadata,
+    passivant: str = "H",
+    bond_length: float = 1.0,
+    coordination_threshold: int = 3,
+    number_of_bonds_to_passivate: int = 1,
+    symmetry_tolerance: float = 0.1,
+    shadowing_radius: float = 2.5,
+    depth: float = 5.0,
+) -> MaterialWithBuildMetadata:
+    analyzer = CoordinationBasedPassivationMaterialAnalyzer(
+        material=material,
+        passivant=passivant,
+        bond_length=bond_length,
+        coordination_threshold=coordination_threshold,
+        number_of_bonds_to_passivate=number_of_bonds_to_passivate,
+        symmetry_tolerance=symmetry_tolerance,
+        shadowing_radius=shadowing_radius,
+        depth=depth,
+    )
+
+    passivant_coordinates = analyzer.get_passivant_coordinates()
+    passivant_configs = [
+        AtomAtCoordinateConfiguration(
+            crystal=material, element=AtomSchema(chemical_element=passivant), coordinate=coord
+        )
+        for coord in passivant_coordinates
+    ]
+    config = PassivationConfiguration(
+        merge_components=[material, *passivant_configs], passivant=passivant, bond_length=bond_length
+    )
+
+    builder = PassivationBuilder()
+    return builder.get_material(config)
+
+
+def get_unique_coordination_numbers(
+    material: MaterialWithBuildMetadata,
+    cutoff: float = 3.0,
+) -> List[int]:
+    """
+    Get the unique coordination numbers for the provided passivation configuration and cutoff radius.
+
+    Args:
+        material (MaterialWithBuildMetadata): The material object.
+        cutoff (float): The cutoff radius for defining neighbors.
+    Returns:
+        set: The unique coordination numbers.
+    """
+    material_with_crystal_sites = MaterialWithCrystalSites.from_material(material)
+    material_with_crystal_sites.analyze()
+    return material_with_crystal_sites.get_unique_coordination_numbers(cutoff=cutoff)
+
+
+def get_coordination_numbers_distribution(
```

Comment:
Should go to `MaterialWithCrystalSites`.

And probably, the MaterialWithCrystalSites itself can be CrystalSitesAnalyzer - perhaps we can revisit this in a cleanup task, maybe put a note.